src/app/pages/question/question.component.ts

```diff
@@ -108,6 +128,12 @@ export class QuestionComponent implements OnInit, CanDeactivate<CanComponentDeac
         }
       ),
     );
+
+    this.router.events.pipe(
+      filter(event => event instanceof NavigationStart)
+    ).subscribe((event: NavigationStart) => {
+      this.nextRoute = event.url;
+    });
```

Comment:
why do we need it?

Reply:
I see, ok. Let's then make this more clear, let's add a method with the proper name, move this logic there and call it here.